### DISCUSSION TITLE
Seeking Alternative Approaches for Resource Provisioning in Perses Helm Chart

### DISCUSSION THREAD

**Discussion by ibakshay** [perses/perses#2399] [Q&A] [answered]
## Current Situation
Currently, [resource provisioning](https://perses.dev/perses/docs/user-guides/provisioning/) (projects, datasources, and dashboards) during Perses application startup requires adding resource manifests [directly to the `datasources` array in the `values.yaml` file](https://github.com/perses/helm-charts/blob/main/charts/perses/values.yaml#L205-L206).

## Problem
This approach presents maintenance challenges:
- Dashboard manifests are typically large, making them difficult to manage within the array
- Adding and updating resource manifests becomes cumbersome
- The current structure isn't scalable for handling multiple dashboards

## Question
Is there a more efficient way to provision resources during Perses startup using the Helm chart, rather than directly embedding them in the [`datasources` array in the values.yaml file](https://github.com/perses/helm-charts/blob/main/charts/perses/values.yaml#L205-L206)?

**Comment by nicolastakashi** [comment]
I think we can use a strategy like Grafana is using: https://github.com/kiwigrid/k8s-sidecar wdyt @ibakshay ?

**Reply by ibakshay** [reply, marked as answer]
Yes, that's a great strategy! I will create a PR in [perses/helm-charts](https://github.com/perses/helm-charts) soon for this :) @nicolastakashi

**Reply by ibakshay** [reply]
@nicolastakashi I have created a PR https://github.com/perses/helm-charts/pull/34 :)
